ic-proxy: enable on master pipelines

pivotal-ning-yu opened the pull request:
- compile gpdb with ic-proxy enabled on master pipelines;
- add the ICW jobs of ic-proxy mode;
- enable ic-proxy on Travis;
- reduce ic-proxy log level;

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

zhangh43 commented on concourse/pipelines/templates/gpdb-tpl.yml:
```diff
@@ -478,6 +486,13 @@ resources:
+  source:
+    bucket: ((gcs-bucket))
+    json_key: ((concourse-gcs-resources-service-account-key))
+    regexp: gp-internal-artifacts/centos7/libuv-(1\.18\.0.*).tar.gz
```
Do we need libuv artifact? I think it should be installed in the docker image.
Also the libuv version supplied by releng team is 1.38.0, but I think we could just use the one in the docker image.

zhangh43 replied:
Could you confirm the libuv artifact? Comments from Kris:
```
The builds are available in GCS with a filename Convention of “libuv-1.38.0.build.[N].tar”, such that [N] is a serial build number:
- gs://pivotal-gpdb-concourse-resources-prod/gp-internal-artifacts/centos7/
- gs://pivotal-gpdb-concourse-resources-prod/gp-internal-artifacts/centos6/
- gs://pivotal-gpdb-concourse-resources-prod/gp-internal-artifacts/ubuntu18.04/
- GCP Project: data-gpdb-ci
```

pivotal-ning-yu replied:
Aha, libuv 1.38 was uploaded to GCP yesterday.  I will update the PR to use this new one.

zhangh43 commented on concourse/pipelines/templates/gpdb-tpl.yml:
```diff
@@ -937,6 +964,44 @@ jobs:
+    file: gpdb_src/concourse/tasks/ic_gpdb.yml
+    image: gpdb7-centos7-test
+    params:
+      MAKE_TEST_COMMAND: -k PGOPTIONS='-c gp_interconnect_type=proxy -c optimizer=on' EXTRA_REGRESS_OPTS=--exclude-tests=external_table,copydml installcheck-world
```
you skip external_table and copydml test in ic-proxy, are they test case issues and feature not supported issues? If it's feature not supported issue, we need record them in the document.

pivotal-ning-yu replied:
They are tracked in an internal story.

pivotal-ning-yu commented:
Pushed a commit to change the GUC `gp_interconnect_proxy_addresses` from a developer option to a normal one, and now we are able to reload this GUC with SIGHUP by executing `gpstop -u`.

pivotal-ning-yu commented:
Upgraded libuv from 1.18.0 to 1.38.0 .

zhangh43 commented on src/backend/cdb/motion/README.ic-proxy.md:
```diff
@@ -606,18 +606,18 @@ the proxy-proxy connections unconditionally.
-   cluster restart;
+When ic-proxy is enabled we need a stage 1.5 between them:
+
+1.1. update the GUC `gp_interconnect_proxy_addresses` to include the new
```
%s/1.1/1.5

pivotal-ning-yu replied:
LOL, good point.

pivotal-ning-yu replied:
Changed.

zhangh43 commented on src/backend/utils/misc/guc_gp.c:
```diff
@@ -4360,7 +4360,7 @@ struct config_string ConfigureNamesString_gp[] =
 #ifdef ENABLE_IC_PROXY
 	{
-		{"gp_interconnect_proxy_addresses", PGC_POSTMASTER, DEVELOPER_OPTIONS,
+		{"gp_interconnect_proxy_addresses", PGC_SIGHUP, GP_ARRAY_TUNING,
```
GP_ARRAY_TUNING  or GP_ARRAY_CONFIGURATION? This GUC more like a configuration.

pivotal-ning-yu replied:
`gp_interconnect_type` is also `GP_ARRAY_TUNING`, but I'm also fine with `GP_ARRAY_CONFIGURATION`. I could change this in next push.

pivotal-ning-yu replied:
Changed to `GP_ARRAY_CONFIGURATION`.

pivotal-ning-yu commented:
Thanks for the approval.  I rebased to latest master and squashed the fixup commits.  Now I'll monitor the devel pipeline, once it's green I'll merge the PR.

pivotal-ning-yu commented:
Hmm, a bad news is libuv 1.38 is only in the gcp prod bucket, the dev bucket only has 1.18, so the PR does not compile on devel pipelines at all.  I'll try to copy the libuv 1.38 packages from prod bucket to dev bucket.

pivotal-ning-yu commented:
I do not have permission to copy packages from prod bucket to dev bucket.  In such a case I have to adjust my plan:
1. wait for Travis builds to pass;
2. merge the PR;
3. monitor the master pipelines;

If there is anything wrong I'll revert the patches.

pivotal-ning-yu commented:
In fact another option is to convert this PR back to use libuv 1.18, which is available on both prod & dev buckets, and it was verified on devel pipeline.  But I think we could first try with the plan with 1.38.

pivotal-ning-yu commented:
OK, the Travis CI is green, I'll merge it now.
